driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -378,6 +406,27 @@ private boolean addConnectionIfUnderMaximum() {
         }
     }
 
+    private PooledConnection tryResurrectFromTrash() {
+        long highestMaxIdleTime = System.currentTimeMillis();
+        PooledConnection chosen = null;
+
+        while (true) {
+            for (PooledConnection connection : trash)
+                if (connection.maxIdleTime > highestMaxIdleTime && connection.maxAvailableStreams() > MIN_AVAILABLE_STREAMS) {
```

Comment:
That makes sense.  At worst the connection will just not be chosen to be used since it does not have least inflight.  So I agree it's fine the way it is.